l1-contracts/contracts/bridge/L1NativeTokenVault.sol

```diff
@@ -0,0 +1,161 @@
+/// @dev Vault holding L1 native ETH and ERC20 tokens bridged into the hyperchains.
+/// @dev Designed for use with a proxy for upgradability.
+contract L1NativeTokenVault is
+    IL1NativeTokenVault,
+    IL1StandardAsset,
+    ReentrancyGuard,
+    Ownable2StepUpgradeable,
+    PausableUpgradeable
+{
+    using SafeERC20 for IERC20;
+
+    /// @dev L1 Shared Bridge smart contract that handles communication with its counterparts on L2s
+    IL1SharedBridge public immutable override L1_SHARED_BRIDGE;
+
+    /// @dev Era's chainID
+    uint256 public immutable ERA_CHAIN_ID;
+
+    /// @dev Maps token balances for each chain to prevent unauthorized spending across hyperchains.
+    /// This serves as a security measure until hyperbridging is implemented.
+    /// NOTE: this function may be removed in the future, don't rely on it!
+    mapping(uint256 chainId => mapping(address l1Token => uint256 balance)) public chainBalance;
+
+    /// @dev A mapping assetInfo => tokenAddress
+    mapping(bytes32 assetInfo => address tokenAddresss) public tokenAddress;
+
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyBridge() {
+        require(msg.sender == address(L1_SHARED_BRIDGE), "NTV not ShB");
+        _;
+    }
+
+    /// @notice Checks that the message sender is the bridgehub.
+    modifier onlyOwnerOrBridge() {
+        require((msg.sender == address(L1_SHARED_BRIDGE) || (msg.sender == owner())), "NTV not ShB or o");
+        _;
+    }
+
+    /// @dev Contract is expected to be used as proxy implementation.
+    /// @dev Initialize the implementation to prevent Parity hack.
+    constructor(IL1SharedBridge _l1SharedBridge, uint256 _eraChainId) reentrancyGuardInitializer {
+        _disableInitializers();
+        ERA_CHAIN_ID = _eraChainId;
+        L1_SHARED_BRIDGE = _l1SharedBridge;
+    }
+
+    /// @dev Initializes a contract for later use. Expected to be used in the proxy
+    /// @param _owner Address which can change L2 token implementation and upgrade the bridge
+    /// implementation. The owner is the Governor and separate from the ProxyAdmin from now on, so that the Governor can call the bridge.
+    function initialize(address _owner) external reentrancyGuardInitializer initializer {
+        require(_owner != address(0), "NTV owner 0");
+        _transferOwnership(_owner);
+    }
+
+    function registerToken(address _l1Token) external onlyOwnerOrBridge {
+        bytes32 tokenInfo = keccak256(abi.encode(address(this), uint256(uint160(_l1Token))));
+        L1_SHARED_BRIDGE.setAssetAddress(bytes32(uint256(uint160(_l1Token))), address(this));
+        tokenAddress[tokenInfo] = _l1Token;
+    }
+
+    /// @notice Allows bridgehub to acquire mintValue for L1->L2 transactions.
+    /// @dev If the corresponding L2 transaction fails, refunds are issued to a refund recipient on L2.
+    function bridgeBurn(
+        uint256 _chainId,
+        uint256 _mintValue,
```

Comment:
Shall we add some logic to actually use the _mintValue?

Reply:
Weth bridge will use mintValue, we do not actually need to use it here I think